ncat/ncat_connect.c

```diff
@@ -1259,6 +1279,11 @@ static void write_socket_handler(nsock_pool nsp, nsock_event evt, void *data)
         ncat_assert(status == NSE_STATUS_SUCCESS);
     }
 
+    if (o.zerobyte){
+      ncat_assert(o.proto == IPPROTO_UDP);
+      nsock_read(nsp, cs.sock_nsi, read_socket_handler, 1000 * 2, NULL);
```

Comment:
We shouldn't hard-code this. We should be able to use either `o.conntimeout` or `o.idletimeout` instead. I lean towards conntimeout because we're waiting for a "connection" instead of a "read," though with UDP they're essentially the same thing.